documentation/docs/customjenkins.md

```diff
@@ -0,0 +1,122 @@
+# Custom Jenkins Setup
+
+Although we recommend to use the Cx server, you don't need to. You can run project "Piper" on your own Jenkins installation. However, you have to care for some settings the Cx Server gives you for free. Furthermore, the support of a none Cx Server installations is challenging. 	
+
+This section describes the necessary adjustments you might have to take.
+
+## Requirements
+
+* Java Runtime Environment 8
+* Installation of Jenkins v 2.60.3 or higher running on Linux. We tested with debian-stretch.
+* A Jenkins user with administration privileges.
+* The Jenkins instance has access to [github.com][github].
+
+## Docker
+Most tools used by project "Piper" to build, test and deploy your application are available as out-of-the box Docker images. No need to manually install them neither on your Jenkins server nor on your Jenkins nodes, and no need to care for the updates. Instead these are pulled from *hub.docker.com* on usage.
+
+Install Docker if you haven't yet. See [Docker Community Edition][docker-install] to install the newest version of Docker.
+**Note:** we have tested on Docker 18.09.6.
+
+If your Jenkins server already runs as Docker container make sure the tools container can run on the Docker host. Extend the Docker call in the following way:
+
+```
+docker run ...  -v /var/run/docker.sock:/var/run/docker.sock ...
+```
+
+## Plugins 
+
+Project "Piper" requires a set of plugins installed on your Jenkins server. This set may evolve in the future. Make sure that all plugins of the appropriate versions are installed.
+
+The Cx server repository contains an [up-to-date list][devops-cxs-plugins] of the required plugins. You could download the list 
+
+```
+curl -o plugins.txt https://raw.githubusercontent.com/SAP/devops-docker-cx-server/master/jenkins-master/plugins.txt
+```
+
+and use the [Jenkins client][jenkins-doc-client] to ease the installation. Run the following command on the Jenkins server with an user holding administrative privileges:
+
+```
+cat plugins.txt | awk '{system("java " "-jar jenkins-cli.jar -s http://localhost:8080 -auth ${ADM_USER}:${ADM_PASSWD} install-plugin " $1)}'
+```
+
+## Shared Library
+
+Shared libraries extending the Jenkins pipeline are defined within the Jenkins system configuration. A library is defined by a link to its source repository and an appropriate version identifier. To add the project "Piper"s library:
+
+1. Open the Jenkins UI under `http://<jenkins-server-address>:<http-port>`, login with administration privileges and choose **Manage Jenkins > Configure System**.
+
+   <p align="center">
+   ![Configure System](images/JenkinsHomeMenuManageConfig.png "Configure System")
+   </p>
+
+1. Scroll down to section **Global Pipeline Libraries** and choose the **Add** button. A new library is created.
```

Comment:
```suggestion
2. Scroll down to section **Global Pipeline Libraries** and choose the **Add** button. A new library is created.
```

Reply:
That's funny, the Eclipse preview is fine. There is no need to increment the index. instead, we need to make sure that the doc generator understands that the list items belong together. the picture in between hinders that. I need to remove the empty line in front of the `<p align="center">`